src/cdk/drag-drop/drop-list-ref.ts

```diff
@@ -419,9 +496,73 @@ export class DropListRef<T = any> {
     });
   }
 
+  /**
+   * Checks whether the user's pointer is close to the edges of either the
+   * viewport or the drop list and starts the auto-scroll sequence.
+   * @param pointerX User's pointer position along the x axis.
+   * @param pointerY User's pointer position along the y axis.
+   */
+  _startScrollingIfNecessary(pointerX: number, pointerY: number) {
+    if (!this.autoScroll) {
+      return;
+    }
+
+    let scrollNode: HTMLElement | Window | undefined;
+    let verticalScrollDirection: AutoScrollVerticalDirection = null;
+    let horizontalScrollDirection: AutoScrollHorizontalDirection = null;
+
+    // @breaking-change 9.0.0 Remove null check for _viewportRuler once it's a required parameter.
+    // Check whether we're in range to scroll the viewport.
+    if (this._viewportRuler) {
+      const {width, height} = this._viewportRuler.getViewportSize();
+      const clientRect = {width, height, top: 0, right: width, bottom: height, left: 0};
+      verticalScrollDirection = getVerticalScrollDirection(clientRect, pointerY);
+      horizontalScrollDirection = getHorizontalScrollDirection(clientRect, pointerX);
+      scrollNode = window;
+    }
+
+    // Otherwise if the window can't be scrolled,
+    // try with the container, if the pointer is close by.
+    if (!verticalScrollDirection && !horizontalScrollDirection &&
+        this._isPointerNearDropContainer(pointerX, pointerY)) {
+      verticalScrollDirection = getVerticalScrollDirection(this._clientRect, pointerY);
+      horizontalScrollDirection = getHorizontalScrollDirection(this._clientRect, pointerX);
+      scrollNode = coerceElement(this.element);
+    }
+
+    // TODO(crisbeto): we also need to account for whether the view or
+    // element are scrollable in the first place.
```

Comment:
It'll still try to scroll it, but it won't go anywhere. I left the TODO for later since there's a case where if there's a scrollable list whose edges overlap with the viewport edge, we'll keep trying to scroll the viewport instead of the list. I decided to leave it out of this PR since it's fairly large already and we'll need some extra logic to ensure that we aren't thrashing the layout when checking this. I've expanded the TODO with more context.